go/proxyd/go.mod

```diff
@@ -1,21 +1,46 @@
 module github.com/ethereum-optimism/optimism/go/proxyd
 
-go 1.16
+go 1.18
```

Comment:
This is going to fail linting in CI until the new `go-builder` image is built, for some reason it didn't publish. Need to debug that

Reply:
This has been resolved, but the linter turns off certain checks for go 1.18 until the packages the linter depends on is updated